[wrangler] Add --json flag to r2 bucket info command

Akshit222 opened the pull request:
Fixes #9433

## What

Adds a `--json` flag to the `wrangler r2 bucket info` command, returning the bucket info in clean JSON format.

## Why

The current output is not machine-readable.  
Adding this flag makes the output usable in scripts and consistent with D1 commands that already support `--json`.  

## How

- Adds a `--json` flag to `r2 bucket info`
- Omits verbose output when `--json` is passed
- Suppresses the Wrangler banner when `--json` is used
- Adds a unit test verifying valid JSON output
- Linting fixed with `pnpm fix`
- Added a changeset as per contribution guidelines

## Test Plan

- ✅ `pnpm exec vitest run packages/wrangler/src/__tests__/r2/bucket.test.ts` passed
- ✅ Manual output verified
- ✅ Verified that banner is suppressed when `--json` is passed

## Example

### With `--json`:
```bash
$ wrangler r2 bucket info my-bucket-name --json

{
  "name": "my-bucket-name",
  "created": "2025-06-07T15:55:22.222Z",
  "location": "APAC",
  "default_storage_class": "Standard",
  "object_count": "0",
  "bucket_size": "0 B"
}
```

### Without `--json`:
```bash
$ wrangler r2 bucket info my-bucket-name

(⛅️ wrangler 4.18.0)
Bucket Name: my-bucket-name  
Created At: 2025-06-07T15:55:22.222Z  
Location: APAC  
Storage Class: Standard  
Object Count: 0  
Size: 0 B
```

---

<!--  
Please don't delete the checkboxes <3  
The following selections do not need to be completed if this PR only contains changes to .md files  
-->

- Tests
  - [x] Tests included
  - [ ] Tests not necessary because:
- Public documentation
  - [x] Cloudflare docs PR(s): https://github.com/cloudflare/cloudflare-docs/pull/24120
  - [ ] Documentation not necessary because: 
- Wrangler V3 Backport
  - [ ] Wrangler PR:
  - [x] Not necessary because: New features are not backported

<!--  
Have you read our [Contributing guide](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md)?  
In particular, for non-trivial changes, please always engage on the issue or create a discussion or feature request issue first before writing your code.  
-->

Akshit222 commented:
Thanks so much for the feedback, @frankichiro — totally agree with you. 
You're right that having the banner in --json mode defeats the purpose of clean, machine-readable output and makes it inconsistent with other commands like D1.

I'll go ahead and update the PR to suppress the banner when --json is used, so the output is clean and parseable out of the box. Expect a commit shortly!

Akshit222 commented:
Just wanted to follow up on this! Let me know if there’s anything else needed from my side to move this forward — happy to make any changes if required. Thanks again for your time!

emily-shen commented:
also, it looks like you have some linting issues. you can run `pnpm fix` to fix them up

Akshit222 commented on packages/wrangler/src/__tests__/r2/bucket.test.ts:
```diff
@@ -0,0 +1,56 @@
+import { requireAuth } from "../../user";
+import { getR2Bucket, getR2BucketMetrics } from "../../r2/helpers";
+
+// Mock the dependencies
```
I've added vi.unmock("../wrangler-banner") to ensure we’re explicitly testing that the banner isn’t printed when using --json.

Akshit222 commented:
Just a quick note: I ended up cleaning up the commit history because the original one had some unrelated files accidentally committed. To keep things clean and focused, I created a fresh branch from main, reapplied only the relevant changes, and force-pushed — which unfortunately removed previous commit-level context. But I’ve addressed all the feedback again in this clean version.

Once you’ve had a chance to review this updated commit, I’ll go ahead and create a follow-up PR to update the Wrangler documentation for the r2 bucket info command as well.

Let me know if anything else is needed!

penalosa commented:
@Akshit222 thanks for this, this is looking pretty good! Would you be able to restore the PR template and fill out the relevant checkboxes, so we can get this merged?

Akshit222 commented:
I've updated the PR description with the template and filled out the checkboxes as requested.
Please let me know if there's anything else you'd like me to address!

petebacondarwin commented:
there are a bunch of linting errors which would mostly be solved by simplifying the test as suggested.

petebacondarwin commented on packages/wrangler/src/r2/bucket.ts:
```diff
@@ -236,7 +248,11 @@ export const r2BucketInfoCommand = createCommand({
 
-		logger.log(formatLabelledValues(output));
+		if (args.json) {
+			console.log(JSON.stringify(output, null, 2));
```
```suggestion
			logger.json(output);
```

petebacondarwin commented:
Please open a draft docs-PR before merging and reference it in the PR description. It can then be reviewed and merged after this had been released.

frankichiro commented:
Wow, this really required much more work than I imagined it would. Thanks @Akshit222 for seeing it through, I really appreciate your efforts for making this work.

Akshit222 commented:
Thanks a lot for the thoughtful feedback and guidance 
I will make the rest of mentioned changes as asked very soon
